File: mobile/lib/dtos/card_learning_stats_update_request.dart
class CardLearningStatsUpdateRequest {
  final String id;
  final double easeFactor;
  final int interval;
  final int repetitions;

  CardLearningStatsUpdateRequest({
    required this.id,
    required this.easeFactor,
    required this.interval,
    required this.repetitions,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'easeFactor': easeFactor,
      'interval': interval,
      'repetitions': repetitions,
    };
  }
}

File: backend/api/DTO/CardLearningStatsUpdateRequest.cs
using System.Text.Json.Serialization;

namespace api.DTO;

public class CardLearningStatsUpdateRequest
{
    public int Id { get; set; }

    public float EaseFactor { get; set; } 
    public int Interval { get; set; }
    public int Repetitions { get; set; }

    public CardLearningStatsUpdateRequest(int id, float easeFactor, int interval, int repetitions)
    {
        Id = id;
        EaseFactor = easeFactor;
        Interval = interval;
        Repetitions = repetitions;
    }
}
